sdk/identity/Azure.Identity/src/Credentials/UsernamePasswordCredential.cs

```diff
@@ -16,6 +16,7 @@ namespace Azure.Identity
     ///  credential will fail to get a token throwing an <see cref="AuthenticationFailedException"/>. Also, this credential requires a high degree of
     ///  trust and is not recommended outside of prototyping when more secure credentials can be used.
     /// </summary>
+    [Obsolete("This credential is deprecated because it doesn't support multifactor authentication (MFA). See https://aka.ms/azsdk/identity/mfa for details about MFA enforcement for Entra ID and migration guidance")]
```

Comment:
```suggestion
    [Obsolete("This credential is deprecated because it doesn't support multifactor authentication (MFA). See https://aka.ms/azsdk/identity/mfa for details about MFA enforcement for Entra ID and migration guidance.")]
```

Reply:
I also like the idea of applying `[EditorBrowsable(EditorBrowsableState.Never)]` to the class so it's less discoverable. Thoughts?